src/downloadmanager.cpp

```diff
@@ -580,7 +598,10 @@ void DownloadManager::startDownload(QNetworkReply *reply, DownloadInfo *newDownl
       newDownload->m_State != STATE_READY &&
       newDownload->m_State != STATE_FETCHINGMODINFO &&
       reply->isFinished()) {
-      downloadFinished(indexByInfo(newDownload));
+      int index = indexByInfo(newDownload);
+      if (index >= 0) {
```

Comment:
The needs for this should be investigated, although probably not relevant for this PR, but maybe add a TODO comment.

Reply:
Is the fix to `downloadFinished` (below) is accepted, this may be redundant (I need to re-check). If it's not accepted, `downloadFinished` may fire with invalid -1 value, causing the `no download index` warning.